src/snapshot/component_map.rs

```diff
@@ -72,55 +69,16 @@ fn apply_rollback_map_to_component_inner<C>(world: &mut World, map: Mut<Rollback
 where
     C: Component + MapEntities,
 {
-    let mut applied_entity_map = map.generate_map();
-
-    EntityMapper::world_scope(&mut applied_entity_map, world, apply_map::<C>);
+    for (original, _new) in map.iter() {
+        if let Some(mut component) = world.get_mut::<C>(original) {
+            component.map_entities(&mut map.as_ref());
+        }
+    }
 
     trace!(
         "Mapped {}",
         bevy::utils::get_short_name(std::any::type_name::<C>())
     );
-
-    // If the entity map is now larger than the set of rollback entities, then dead entities were created.
-    // TODO: This workaround is required because the current behavior of `map_all_entities` is to change all entities,
-    // creating dead entities instead of leaving them with their original value. If `EntityMapper` behavior changes,
-    // then this workaround may no longer be required.
```

Comment:
Begone! :D

Reply:
This is my favourite part.

Reply:
Yeah it felt _really_ good to make an update PR that _removed_ more code than it added, and got us a little extra performance as well! (I assume, haven't measured, and it's probably not much anyway)